tests/conftest.py

```diff
@@ -546,3 +546,102 @@ def _dir_sample_patches(_sample_patch1, _sample_patch2, tmpdir_factory):
+
+    if not pathlib.Path(patch_file_path).is_file():
+        print("\nDownloading sample patch 2")
+        r = requests.get(
+            "https://tiatoolbox.dcs.warwick.ac.uk/testdata/models/"
+            "CRC-Prim-HE-07_APPLICATION.tif"
+        )
+        with open(patch_file_path, "wb") as f:
+            f.write(r.content)
+    else:
+        print("\nSkipping Source Image")
+
+    return patch_file_path
+
+
+@pytest.fixture(scope="session")
+def _mini_wsi1_svs(tmpdir_factory):
+    """Sample pytest fixture for torch wsi dataset.
+    Download svs image for pytest.
+
+    """
+    svs_file_path = tmpdir_factory.mktemp("data").join("CMU-mini.svs")
+    if not pathlib.Path(svs_file_path).is_file():
+        print("\nDownloading %s" % svs_file_path)
+        r = requests.get(
+            "https://tiatoolbox.dcs.warwick.ac.uk/testdata/models/CMU-mini.svs"
+        )
+        with open(svs_file_path, "wb") as f:
+            f.write(r.content)
+    else:
+        print("\nSkipping %s" % svs_file_path)
+
+    return svs_file_path
+
+
+@pytest.fixture(scope="session")
+def _mini_wsi2_jp2(tmpdir_factory):
+    """Sample pytest fixture for torch wsi dataset.
+    Download svs image for pytest.
+
+    """
+    svs_file_path = tmpdir_factory.mktemp("data").join("mini_wsi2.jp2")
+    if not pathlib.Path(svs_file_path).is_file():
+        print("\nDownloading %s" % svs_file_path)
+        r = requests.get(
+            "https://tiatoolbox.dcs.warwick.ac.uk/testdata/models/mini_wsi2.jp2"
+        )
+        with open(svs_file_path, "wb") as f:
+            f.write(r.content)
+    else:
+        print("\nSkipping %s" % svs_file_path)
+
+    return svs_file_path
+
+
+@pytest.fixture(scope="session")
+def _mini_wsi1_jpg(tmpdir_factory):
+    """Sample pytest fixture for torch wsi dataset.
+    Download svs image for pytest.
+
+    """
+    svs_file_path = tmpdir_factory.mktemp("data").join("CMU-mini.jpg")
+    if not pathlib.Path(svs_file_path).is_file():
+        print("\nDownloading %s" % svs_file_path)
+        r = requests.get(
+            "https://tiatoolbox.dcs.warwick.ac.uk/testdata/models/CMU-mini.jpg"
+        )
+        with open(svs_file_path, "wb") as f:
+            f.write(r.content)
+    else:
+        print("\nSkipping %s" % svs_file_path)
+
+    return svs_file_path
+
+
+@pytest.fixture(scope="session")
+def _mini_wsi1_msk(tmpdir_factory):
+    """Sample pytest fixture for torch wsi dataset.
+    Download svs image for pytest.
+
+    """
+    file_path = tmpdir_factory.mktemp("data").join("CMU-mask.png")
+    if not pathlib.Path(file_path).is_file():
+        print("\nDownloading %s" % file_path)
+        r = requests.get(
+            "https://tiatoolbox.dcs.warwick.ac.uk/testdata/models/CMU-mask.png"
+        )
+        with open(file_path, "wb") as f:
+            f.write(r.content)
+    else:
+        print("\nSkipping %s" % file_path)
+
+    return file_path
```

Comment:
Just noticed that these are also using the old python string interpolation. Can we use f-strings here to be consistent and I think they are recommended over the old method.

Reply:
I think I may need a seperate PR to clean all of these format str. It spreads out all over the models code too.